tools/state-viewer/src/cli.rs

```diff
@@ -679,20 +718,79 @@ pub struct ViewTrieCmd {
     /// For format=pretty this measures depth in terms of key nibbles.
     #[clap(long)]
     max_depth: u32,
+    /// Limits how many entries are printed to the output.
+    #[clap(long)]
+    limit: Option<u32>,
+    /// Filters output to only show records of given type.
+    #[clap(long)]
+    rtype: Option<StateRecordType>,
```

Comment:
nit: full variable names please (I'm fine with either type, record_type or state_record_type)

Reply:
`type` is Rust reserved keyword, I will change it to `kind`